### ISSUE TITLE
Text Input typing bug

### ISSUE THREAD

**Issue by Neizvestnyj** [kivy/kivy#7415]
* Python: 3.8
* OS: android
* Kivy: master
* Kivy installation method: pip

When you enter text in the text field, after pressing the space bar, it disappears somewhere, if you start erasing it, it appears, but only after erasing the visible characters.

https://user-images.githubusercontent.com/40869738/110664448-745b6d80-81d8-11eb-8b8c-928131a487bc.mp4

**Comment by darpan5552** [comment]
https://youtu.be/OCk6OuXq4dA explains the problem.

**Comment by Neizvestnyj** [comment]
> https://youtu.be/OCk6OuXq4dA explains the problem.

As I understand this is a bug, but how to fix it?

**Comment by Neizvestnyj** [comment]
> https://youtu.be/OCk6OuXq4dA explains the problem. This is just a reference. Please try that app and report if issue is exactly the same. I have tried that and can confirm that when using keyboard via otg, the problem mentioned in video disappears

I downloaded the app from the video above, the bug is the same

**Comment by Neizvestnyj** [comment]
@darpan5552 
Uploading Screenrecorder-2021-03-11-21-23-40-771(0).mp4…

**Comment by Neizvestnyj** [comment]
https://user-images.githubusercontent.com/40869738/110835925-7abd1880-82b0-11eb-8fcb-5c3bdb76d0f0.mp4

**Comment by darpan5552** [comment]
Upon close observation, we can observe that kivy is forgetting its cursor position. Suppose we are at `position1` to type 4 characters, then for typing 5th character if we go to `position2`, all the 4 characters including 5th character comes at `position2`.

**Comment by Neizvestnyj** [comment]
@darpan5552 Yes, this is a textinput bug, I looked at the source code, but I still can't figure out where the error is

**Comment by darpan5552** [comment]
@Neizvestnyj May be the bug is in kivy.core.text instead of kivy.uix.textinput

**Comment by Neizvestnyj** [comment]
@darpan5552 Don't know (

**Comment by tshirtman** [comment]
i can confirm weird behaviors, it still kind of work, if you trust it to do the right thing and continue, but this is not ideal.
https://user-images.githubusercontent.com/22759/112563046-e89d3000-8dd8-11eb-949b-840e2a2aed66.mp4

**Comment by olek5andr** [comment]
Hi, I'm sorry for possible spaming, but I'm having a relatable problem https://github.com/kivy/kivy/issues/7545. 
Have been able to find some kind of workaround?

**Comment by olek5andr** [comment]
> * Python: 3.8
> * OS: android
> * Kivy: master
> * Kivy installation method: pip
> 
> When you enter text in the text field, after pressing the space bar, it disappears somewhere, if you start erasing it, it appears, but only after erasing the visible characters.
> 
>  bug.mp4

Hi, I'm sorry for possible spaming, but I'm having a relatable problem #7545.
Have been able to find some kind of workaround?

**Comment by Neizvestnyj** [comment]
> > * Python: 3.8
> > * OS: android
> > * Kivy: master
> > * Kivy installation method: pip
> > 
> > When you enter text in the text field, after pressing the space bar, it disappears somewhere, if you start erasing it, it appears, but only after erasing the visible characters.
> > bug.mp4
> 
> Hi, I'm sorry for possible spaming, but I'm having a relatable problem #7545.
> Have been able to find some kind of workaround?

No, it doesn't work at the moment

**Comment by darpan5552** [comment]
@olek5andr 
Use kivy==2.0.0
I have observed that it's working much better

**Comment by Neizvestnyj** [comment]
Yes, at the moment the problem is still being solved

**Comment by olek5andr** [comment]
> @olek5andr
> Use kivy==2.0.0
> I have observed that it's working much better

Using kivy == 2.0.0 is not an option, because when using kivy=2.0.0 with kivymd it is lagging drastically on an Android device. It appears to run much smoother when kivy==master, but then this issue with TextInput arises.
I'm trying to identify the specific commit after which it started, wasn't able to find it yet...

**Comment by darpan5552** [comment]
@olek5andr 
I am using kivy==2.0.0 with kivymd==0.104.1 and still haven't observed any lagging. May be those application are lot simpler.
https://github.com/kengoon/PyJniusTester
https://github.com/dcindia/matrix-calculator
https://github.com/dcindia/KivyAppUpdater

**Comment by ChihayaK** [comment]
It seems the problem still exists in version 2.0.0. Also I had encountered another bug that might related to this issue: When pasting long text using gboard auto paste feature the pasted text is incomplete.

**Comment by hussam132** [comment]
any updates !👀

**Comment by Philipmikmik** [comment]
Can someone test if the bug persists if using p4a manually, since i tried it and it was fixed

**Comment by hussam132** [comment]
@Philipmikmik what you mean exactly

**Comment by Philipmikmik** [comment]
@hussam132 well i was working on a app that experienced this error and now after some time i tried python-for-android by itself, but i also changed a lot of things like my os, python version etc. in the time passed and cannot confirm it was that that fixed it. So im asking if someone can test it

**Comment by Philipmikmik** [comment]
@hussam132 im sorry but i dont think thats possible, since its been over half a year. And i also had a problem that made buildozer unusable thats why i used p4a, its possible the error got cleared in that time

**Comment by rqsrdco** [comment]
Has anyone fixed it yet? (TextInput)

**Comment by nxtsoul** [comment]
Any updates ? I'm using kivy version 2.0.0 and the bug still appearing.

**Comment by Neizvestnyj** [comment]
> Any updates ? I'm using kivy version 2.0.0 and the bug still appearing.

No, in 2.1.0 this feature was removed

**Comment by Drohne1** [comment]
Python: 3.10.4
OS: android
Kivy: 2.1.0 master
Kivymd: 1.0.2
buildozer: 1.4.0
p4a: 2022.7.20
Kivy installation method: pip (pipenv)

I'm facing the same behavior as @Neizvestnyj described in his/her first post. 
Is anybody working on that issue or is a workaround available? @tshirtman Thank you for every small hint.

**Comment by OrangMutan** [comment]
I saw this behavior already in this environment:
Python: 2.7.2
OS: android
Kivy: 1.9.0
Problem wasn't there at least until android-version 7.1.1, it appeared with newer devices (don't know exactly when it first come up).

For some other reason I upgrades everything:
Python: 3.9.9
OS: android
Kivy: 2.1.0

I was hoping in the meantime this bug has been found and eliminated. But not :( 
From my observation there must be some connection to the android-version!
Any hint would be very welcome ...

**Comment by OrangMutan** [comment]
I wanted to verify, if my statement above is true: I installed new apk (Python: 3.9.9, Kivy: 2.1.0) on my old Sony Z4 with android 7.1.1, and - contradicting to what I said -  bug in TextInput occurs. So: There is *no* obvious connection to the version of android.
Would like to help in finding a solution - but don't know how ...
